crypto/box/src/versioned.rs

```diff
@@ -210,8 +212,11 @@ mod test {
                 for _reps in 0..50 {
                     let ciphertext = algo.encrypt(&mut rng, &a_pub, plaintext).unwrap();
                     let decrypted = algo.decrypt(&not_a, &ciphertext);
-                    assert!(decrypted.is_err());
-                    assert_eq!(decrypted, Err(Error::MacFailed));
+                    if decrypted.is_err() {
+                        assert_eq!(decrypted, Err(Error::MacFailed));
```

Comment:
i think we should just remove the `MacFailed` error type if it will never be signaled that way, and will use the bool instead?